eth/tracers/logger/logger.go

```diff
@@ -78,12 +78,12 @@ type StructLog struct {
 
 // overrides for gencodec
 type structLogMarshaling struct {
-	Gas         math.HexOrDecimal64
-	GasCost     math.HexOrDecimal64
-	Memory      hexutil.Bytes
-	ReturnData  hexutil.Bytes
-	OpName      string `json:"opName"` // adds call to OpName() in MarshalJSON
-	ErrorString string `json:"error"`  // adds call to ErrorString() in MarshalJSON
+	Gas         math.HexOrDecimal64 `json:",omitempty"`
+	GasCost     math.HexOrDecimal64 `json:",omitempty"`
```

Comment:
I think this will cause `0` to become omitted, which is technically incorrect, IMO. So I don't think we should set `omitEmpty` for `Gas` and `GasCost`.